pennylane/templates/embeddings/qaoa.py

```diff
@@ -260,22 +259,38 @@ def circuit(weights, f=None):
         1-dimensional Ising model.
 
     """
-    wires = Wires(wires)
-    repeat = _preprocess(features, wires, weights)
-
-    if local_field == "Z":
-        local_fields = RZ
-    elif local_field == "X":
-        local_fields = RX
-    elif local_field == "Y":
-        local_fields = RY
-    else:
-        raise ValueError(f"did not recognize local field {local_field}")
 
-    for l in range(repeat):
-        # apply alternating Hamiltonians
-        qaoa_feature_encoding_hamiltonian(features, wires)
-        qaoa_ising_hamiltonian(weights[l], wires, local_fields)
+    num_params = 2
+    num_wires = AnyWires
+    par_domain = "A"
+
+    def __init__(self, features, weights, wires, local_field="Y", do_queue=True):
+
+        wires = Wires(wires)
+
+        _preprocess(features, wires, weights)
```

Comment:
Since templates are now a class, it probably makes sense to move `_preprocess` as a static method, for code organization :)